limitedlistener.go

```diff
@@ -74,10 +77,16 @@ func (sl *limitedListener) Restart() {
 type LimitedConn struct {
 	net.Conn
 	counter *uint64
+	closed  uint32
 }
 
 func (c *LimitedConn) Close() (err error) {
+	if atomic.SwapUint32(&c.closed, 1) == 1 {
```

Comment:
Oh, I didn't think of the case Close() being called more than once! But indeed it happens! Thanks